orderer/consensus/hlmirbft/chain.go

```diff
@@ -1467,8 +1467,36 @@ func (c *Chain) Apply(*msgs.QEntry) error {
 	return nil
 }
 
+//JIRA FLY2-66 proposed changes: Implemented the Snap Function
 func (c *Chain) Snap(networkConfig *msgs.NetworkState_Config, clientsState []*msgs.NetworkState_Client) ([]byte, []*msgs.Reconfiguration, error) {
-	return nil, nil, nil
+
+	var persistBytes []byte
+	c.Node.PurgeSnapFiles(c.opts.SnapDir)
+	pr := c.Node.PendingReconfigurations
+	c.Node.PendingReconfigurations = nil
+
+	c.Node.CheckpointState = &msgs.NetworkState{
+		Config:                  networkConfig,
+		Clients:                 clientsState,
+		PendingReconfigurations: pr,
+	}
+	c.Node.CheckpointHash = c.Node.ActiveHash.Sum(nil)
+	c.Node.ActiveHash = c.Node.Hasher.New()
+	c.Node.ActiveHash.Write(c.Node.CheckpointHash)
+	checkpointBytes, err := proto.Marshal(c.Node.CheckpointState)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	persistBytes = append([]byte{}, c.Node.CheckpointHash...)
+	persistBytes = append(persistBytes, checkpointBytes...)
+	err = c.Node.PersistSnapshot(c.Node.CheckpointSeqNo, persistBytes)
```

Comment:
Should CheckpointSeqNo increment each time a Snap is called? Currently it doesn't change.

Reply:
Fixed the issue